content/en/docs/pipelines/installation/localcluster-deployment.md

```diff
@@ -0,0 +1,216 @@
++++
+title = "Kubeflow Pipelines Local Cluster Deployment"
+description = "Instructions to deploy Kubeflow Pipelines standalone to a local cluster for testing purposes"
+weight = 20
++++
+
+As an alternative to deploying Kubeflow Pipelines (KFP) as part of the
+[Kubeflow deployment](/docs/started/getting-started/#installing-kubeflow), you also have a choice
+to deploy only Kubeflow Pipelines. Follow the instructions below to deploy
+Kubeflow Pipelines as part of your local environment for test purposes using the supplied kustomize manifests.
+
+You should be familiar with [Kubernetes](https://kubernetes.io/docs/home/),
+[kubectl](https://kubernetes.io/docs/reference/kubectl/overview/), and [kustomize](https://kustomize.io/).
+
+{{% alert title="Installation options for Kubeflow Pipelines standalone" color="info" %}}
+This guide currently describes how to install Kubeflow Pipelines standalone
+on Kind,K3s and K3s on WSL2 other configurations will be added here covering minikube(TBD) and others.
+{{% /alert %}}
+
+## Before you get started
+
+Working with Kubeflow Pipelines Standalone requires a Kubernetes cluster as well as an installation of kubectl.
+
+### Download and install kubectl
+
+Download and install kubectl by following the [kubectl installation guide](https://kubernetes.io/docs/tasks/tools/install-kubectl/).
+
+You need kubectl version 1.14 or higher for native support of kustomize.
+
+### Set up your cluster on Kind
+
+kind is a tool for running local Kubernetes clusters using Docker container “nodes”.
+kind was primarily designed for testing Kubernetes itself, but may be used for local development or CI.
+See the guide [to install and configure kind](https://kind.sigs.k8s.io/docs/user/quick-start/).
+For simplicity we report below the basic steps here.
```

Comment:
I'd rather prefer keeping third-party complex content away from kubeflow.org:
1. Kubeflow cannot help users debug these steps
1. Kind, k3c, ... haven't reach 1.0 yet, these may change, and we don't want to spend extra efforts maintaining these when they change

I'd prefer only necessary information that's either:
* specific for Kubeflow Pipelines
* important to know and not easy to add a link to 

what do you think?

Reply:
Hey, thanks for the comment...I agree with the general view, my intent here was to add the minimal steps (that are enough mature) to install a cluster. The commands reported are in the products since the first release so should be consistent over time. The reason why I did this was based on my early experience when I approached the "how to install" guide on dev environments and I had to jump on and off from other guides losing often track on what I was doing here and there... happy to drop the steps and leave the link to the guide but I think that for those who want quickly start the effort of maintaining the minimal steps pay off to make the life of the readers easier.
But as said happy to drop those lines anyways

Reply:
I understand the pain for having to jump around.
For items you think are stable (same starting from first version), I'm okay we keep it.

You definitely are in a better position to make these trade offs.

Reply:
I can maintain them by the time being so let's leave them like this...it would be great overall to do a recurring survey on users on how to improve documentation over time if it is not yet done..I could help here a Product Manager (maybe in the PM sig?)

Reply:
@alfsuse @Bobgy Whatever you decide on the third party, such as `kind`, let me know.

Reply:
_(I'm sharing my refactored section on `kind` here, in case you want to use it.)_

_(Feel free to change this.)_

_(Because I can't have "nested" code blocks in Markdown (three back ticks) without breaking the preview here, add them before and after command lines, if necessary.)_

```
## Installing kind

[kind](https://kind.sigs.k8s.io) is a tool for running local Kubernetes clusters using Docker 
container nodes. `kind` was primarily designed for testing Kubernetes itself, but may be 
used for local development or CI.

You can install and configure kind by following the [official quick start](https://kind.sigs.k8s.io/docs/user/quick-start/).

Below are basic installation steps to get you started.

**On Linux:**

Download and move the `kind` executable to your directory in your PATH by running the 
following commands:

```shell
  curl -Lo ./kind https://kind.sigs.k8s.io/dl/{KIND_VERSION}/kind-linux-amd64 
  chmod +x ./kind
  mv ./kind /{YOUR_KIND_DIRECTORY}/kind
```

```
where:

* `{KIND_VERSION}`: the kind version; for example, `v0.8.1` as of the date this guide was written
* `{YOUR_KIND_DIRECTORY}`: your directory in PATH

**On macOS:**

Use [Homebrew](https://brew.sh) by running the following command:

```
```
brew install kind
```

```
**On Windows:**

- You can use the administrative PowerShell console to run the following commands to 
download and move the `kind` executable to a directory in your PATH:

```
```
curl.exe -Lo kind-windows-amd64.exe https://kind.sigs.k8s.io/dl/{KIND_VERSION}/kind-windows-amd64
Move-Item .\kind-windows-amd64.exe c:\{YOUR_KIND_DIRECTORY}\kind.exe
```

```
where:

* `{KIND_VERSION}`: the kind version; for example, `v0.8.1` as of the date this guide was 
written
* `{YOUR_KIND_DIRECTORY}`: your directory in PATH

- Alternatively, you can use Chocolatey [https://chocolatey.org/packages/kind](https://chocolatey.org/packages/kind):

```
```
choco install kind
```
```

**Note:** kind use containerd as default container-runtime hence you cannot use the 
standard kubeflow pipeline manifests.

**References**:

* [kind Quick Start Guide](https://kind.sigs.k8s.io/docs/user/quick-start/)

* [kind: Known Issues](https://kind.sigs.k8s.io/docs/user/known-issues/)

* [kind: Working Offline](https://kind.sigs.k8s.io/docs/user/working-offline/)

## Creating a cluster on kind

Having installed kind, you can create a Kubernetes cluster on kind by running a simple 
command:

```
```
kind create cluster
```

```

This will bootstrap a Kubernetes cluster using a pre-built node image. You can find that 
image on the Docker Hub `kindest/node` [here](https://hub.docker.com/r/kindest/node). 
If you wish to build the node image yourself, you can use the `kind build node-image` 
command—see the official [building image](https://kind.sigs.k8s.io/docs/user/quick-start/#building-images) 
section for more details. And, to specify another image use the `--image` flag.

By default, the cluster will be given the name kind. Use the `--name` flag to assign the 
cluster a different context name.
```